### PR TITLE
Persist cheEditor and kubernetes components when updating the devfile

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### What does this PR do?
- Fixes updating of the devfile by a request from Che-Theia.
- Fixes setting of the plugins to the devfile form the Plugins panel.

Link with https://github.com/eclipse-che/che-theia/pull/1076

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

### CODE REVIEW

File: extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-devfile-service-impl.ts [hunk 1]
```diff
@@ -553,13 +553,25 @@ export class CheServerDevfileServiceImpl implements DevfileService {
   }
 
   devfileV1toDevfileV2(devfileV1: cheApi.workspace.devfile.Devfile): Devfile {
+    const components: DevfileComponent[] = [];
+    if (devfileV1.components) {
+      devfileV1.components.forEach(component => {
+        if (component.type === 'cheEditor' || component.type === 'kubernetes') {
```
Comment (benoitf) [inline extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-devfile-service-impl.ts:559]: I don't see why we filter out these components ?

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-devfile-service-impl.ts:559]: Giving them to the client does not have any sense, as it uses devfile2.

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-devfile-service-impl.ts:559]: Do you prefer to sent them to the client and mark with special attribute?

**Comment by vitaliy-guliy** [comment]
There is another approach has been merged to master https://github.com/eclipse-che/che-theia/pull/1084
Another PR for kubernetes components wil be provided within https://github.com/eclipse/che/issues/19634

**Pull request closed without merging**